cmd/utils.go

```diff
@@ -182,26 +183,64 @@ func contains(slice interface{}, elem interface{}) bool {
 	return false
 }
 
+// profilerWrapper is created becauses pkg/profiler doesn't
+// provide any API to calculate the profiler file path in the
+// disk since the name of this latter is randomly generated.
+type profilerWrapper struct {
+	stopFn func()
+	pathFn func() string
+}
+
+func (p profilerWrapper) Stop() {
+	p.stopFn()
+}
+
+func (p profilerWrapper) Path() string {
+	return p.pathFn()
+}
+
 // Starts a profiler returns nil if profiler is not enabled, caller needs to handle this.
-func startProfiler(profiler string) interface {
+func startProfiler(profilerType, dirPath string) interface {
 	Stop()
+	Path() string
 } {
+
+	var err error
+	if dirPath == "" {
+		dirPath, err = ioutil.TempDir("", "profile")
+		logger.FatalIf(err, "Unable to create a temporary directory for profiling")
+	}
+
+	var profiler interface {
+		Stop()
+	}
+
 	// Enable profiler if ``_MINIO_PROFILER`` is set. Supported options are [cpu, mem, block].
```

Comment:
This comment looks out of place here as we are not checking any environment variables near by this code.

Reply:
> This comment looks out of place here as we are not checking any environment variables near by this code.

Fixed